enterprise/users/src/main/java/io/crate/metadata/UsersPrivilegesMetaData.java

```diff
@@ -65,16 +66,17 @@ public static UsersPrivilegesMetaData copyOf(@Nullable UsersPrivilegesMetaData o
         for (Map.Entry<String, Set<Privilege>> entry : oldMetaData.usersPrivileges.entrySet()) {
             Set<Privilege> privileges = new HashSet<>(entry.getValue().size());
             for (Privilege privilege : entry.getValue()) {
-                if (privilege.clazz().equals(Privilege.Clazz.TABLE) == false) {
-                    privileges.add(new Privilege(privilege.state(), privilege.type(), privilege.clazz(),
-                        privilege.ident(), privilege.grantor()));
+                PrivilegeIdent privilegeIdent = privilege.ident();
+                if (privilegeIdent.clazz().equals(Privilege.Clazz.TABLE) == false) {
+                    privileges.add(new Privilege(privilege.state(), privilegeIdent.type(), privilegeIdent.clazz(),
```

Comment:
pls create a constructor with the PrivilegeIdent and reuse privilegeIdent, however i am not sure if we need targetIdent here?